Cogs.Publishers/JsonScehmaPropConverter.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using Newtonsoft.Json.Linq;
+
+namespace Cogs.Publishers
+{
+    internal class JsonScehmaPropConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return objectType == typeof(List<JsonSchemaProp>);
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            if (value.GetType() == typeof(List<JsonSchemaProp>))
```

Comment:
In c# 7 there is a new feature that makes this type of type testing easier called pattern matching
https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7#pattern-matching

We can change the if statement to 
if(value is List<JsonSchemaProp>prop_list) { .... } to test the type and declare a typed variable at the same time